spec/requests/middleware_datasources_spec.rb

```diff
@@ -0,0 +1,68 @@
+describe 'Middleware Datasources API' do
+  let(:datasource) { FactoryGirl.create(:middleware_datasource) }
+
+  # For some reason middleware_datasources_url is not returning the full
+  # url, just the path portion. This is a hack, but will do the trick.
+  def datasource_url
+    id = ApplicationRecord.compress_id(datasource.id)
+    "http://www.example.com#{middleware_datasources_url(id)}"
+  end
+
+  def match_id
+    eq ApplicationRecord.compress_id(datasource.id).to_s
+  end
+
+  describe '/' do
+    it 'forbids access without an appropriate role' do
+      api_basic_authorize
+
+      run_get middleware_datasources_url
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'returns an empty listing of datasources' do
+      api_basic_authorize collection_action_identifier(:middleware_datasources, :read, :get)
+
+      run_get middleware_datasources_url
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to eq(
+        'name'      => 'middleware_datasources',
+        'count'     => 0,
+        'resources' => [],
+        'subcount'  => 0
+      )
+    end
+
+    it 'returns a a listing of datasources' do
+      datasource
+
+      api_basic_authorize collection_action_identifier(:middleware_datasources, :read, :get)
+
+      run_get middleware_datasources_url
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to eq(
+        'name'      => 'middleware_datasources',
+        'count'     => 1,
+        'resources' => [{
+          'href' => datasource_url
```

Comment:
You could use `a_string_matching(middleware_datasources_url(datasource.compressed_id)` if you didn't want to write your own helper methods (this is pretty much the strategy we use elsewhere), but this is fine as is.

Reply:
Ok, I'll leave it like this. After your change to use the actual rails routes for that I can remove them.